components/dashboard/src/components/WorkspaceLogs.tsx

```diff
@@ -56,7 +57,7 @@ export default function WorkspaceLogs({ logsEmitter, errorMessage, classes, xter
         terminal.loadAddon(fitAddon);
         terminal.open(xTermParentRef.current);
 
-        let logBuffer = "";
+        let logBuffer = new Uint8Array();
         let isWriting = false;
 
         const processNextLog = () => {
```

Comment:
Right below here on L66-67, thanks to now using a `Uint8Array` we can replace `.slice` with `.subarray` to:
1. Share the same memory and not copy data over
2. Make it more fun

```ts
            const logs = logBuffer.subarray(0, MAX_CHUNK_SIZE);
            logBuffer = logBuffer.subarray(logs.length);
```

Reply:
Ah, yes, let's do that!